File: prototype-pattern/java/src/Circle.java
package src;

// 具体原型类，克隆方法会创建一个新对象并将其传递给构造函数。
public class Circle extends Shape {
  public Circle() {
    super();
    type = "Circle";
  }

  @Override
  public void draw() {
    System.out.println("Circle::draw() method.");
  }
}

File: prototype-pattern/go/src/Circle.go
package src

import "fmt"

type Circle struct {
  // 匿名方式聚合父类
  ShapeBase
}

func (c *Circle) Draw() {
  fmt.Println("Circle::Draw()")
}

func (c *Circle) SetColor(color string) {
  c.color = color
}
